kernel/gic/lib.rs

```diff
@@ -0,0 +1,9 @@
+//! Allows configuring the Generic Interrupt Controller
+#![no_std]
+#![feature(doc_cfg)]
+
+#[cfg(any(target_arch = "aarch64", doc))]
+mod src;
+
+#[cfg(any(target_arch = "aarch64", doc))]
+pub use src::*;
```

Comment:
> If I build this PR for x86 as normal, what happens?

Thanks to this setup, this crate appears as empty during an x86_64 build. As you can see, it keeps the standard src folder for the real code, and the top-level lib.rs is just an arch gate.

Reply:
that's good, but you can still achieve that with the conventional source code organization that i mentioned above, right?

Reply:
Yep, will do as you prefer

Reply:
Either one of the above structures is fine, including the latter. We just want all the source code to be inside `src/`. I suppose my main critique of the latter option is that `aarch64` is a vague name for a module, especially since the entire GIC crate is already aarch64-specific.

We don't need "include_str" here, that's not appropriate for Rust source code. Target-specific `mod` statements are preferred to include/exclude code.

Reply:
> We don't need "include_str" here, that's not appropriate for Rust source code. Target-specific `mod` statements are preferred to include/exclude code.

Well that excludes the second solution, as far as I know you can't have multiple levels of modules in the same folder.

Ok, I'll add another directory.

Reply:
I just pushed a commit that migrates to this layout:
```
kernel/gic
├── Cargo.toml
└── src
    ├── lib.rs
    └── gic
        ├── cpu_interface_gicv2.rs
        ├── cpu_interface_gicv3.rs
        ├── dist_interface.rs
        ├── mod.rs
        └── redist_interface.rs
```